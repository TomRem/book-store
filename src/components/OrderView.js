import React, { Component } from 'react'

export class OrderView extends Component {
    render() {
        return (
            <div>
                <span>{this.props.book.name}</span>
                <button className="btn btn-danger" onClick={(event) => this.props.removeFromOrder(this.props.book.name)}>
                    Usu≈Ñ</button>
                <hr />
            </div>
        )
    }
}

export default OrderView
