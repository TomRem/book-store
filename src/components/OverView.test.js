import React from 'react';
import ReactDOM from 'react-dom';
import OrderView from './OrderView';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({
    adapter: new Adapter()
});

describe('OrderView tests', () => {

    it('Snapshot matches', () => {
        const book = {
            name: 'Book Title'
        }
        const wrapper = shallow(<OrderView book={book} />);
        expect(wrapper).toMatchSnapshot();
    })

    it('Props and title of book is the same', () => {
        const book = {
            name: 'Book Title'
        }
        const wrapper = shallow(<OrderView book={book} />);
        expect(wrapper.find('div').find('span').text()).toBe(book.name);
    })
})