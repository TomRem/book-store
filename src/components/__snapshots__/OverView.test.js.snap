// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderView tests Snapshot matches 1`] = `ShallowWrapper {}`;
